code/Model/SiteTree.php

```diff
@@ -3248,25 +3256,24 @@ public function getPageIconURL()
      */
     public function classDescription()
     {
+        // Ensure base class has an appropriate description if not explicitly set,
+        // since we can't set that config for projects.
         $base = in_array(static::class, [Page::class, SiteTree::class]);
         if ($base) {
-            return $this->config()->get('base_description');
+            $baseDescription = static::config()->get('base_class_description');
+            // Fall back to the deprecated config
+            if (!$baseDescription) {
+                $baseDescription = static::config('base_description');
+            }
+            return $baseDescription;
         }
-        return $this->config()->get('description');
-    }
-
-    /**
-     * Get localised description for this page
-     *
-     * @return string|null
-     */
-    public function i18n_classDescription()
```

Comment:
`i18n_classDescription()` is removed here but will now exist on a parent class, so no need to deprecate.